api/PayrollProcessor.Core.Domain/Intrastructure/Identifiers/EntityIdGenerator.cs

```diff
@@ -0,0 +1,14 @@
+using System;
+
+namespace PayrollProcessor.Core.Domain.Intrastructure.Identifiers
+{
+    public interface IEntityIdGenerator
+    {
+        Guid Generate();
+    }
+
+    public class EntityIdGenerator : IEntityIdGenerator
```

Comment:
This will let us be able test methods that perform Guid creation a little easier.